addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsParam.java

```diff
@@ -121,6 +124,7 @@ public List<CustomPayload> getPayloads() {
     public void setPayloads(List<CustomPayload> payloads) {
         Map<String, List<CustomPayload>> newPayloads =
                 payloads.stream()
+                        .filter(pl -> pl.getCategory() != null)
```

Comment:
Can this still happen? Be good to have tests that cover both this and the above change.

Reply:
I guess this can only happen if someone edits the config file manually.

Reply:
But aren't we skipping those categories when loading the configs?

Reply:
Yup, you're right. The earlier change handles it. Now that all these items are addressed.